api/repositories/route_repository.go

```diff
@@ -141,6 +140,24 @@ func (f *RouteRepo) GetRoute(ctx context.Context, authInfo authorization.Info, r
 	}
 
 	toReturn, err := returnRoute(cfRouteList.Items)
+	if err != nil {
+		return RouteRecord{}, err
+	}
+
+	userClient, err := f.userClientFactory.BuildClient(authInfo)
+	if err != nil {
+		return RouteRecord{}, fmt.Errorf("get-route failed to build user client: %w", err)
+	}
+
+	err = userClient.Get(ctx, client.ObjectKey{Namespace: toReturn.SpaceGUID, Name: toReturn.GUID}, &networkingv1alpha1.CFRoute{})
+
+	if err != nil {
+		if apierrors.IsForbidden(err) {
+			return RouteRecord{}, NewForbiddenError(RouteResourceType, err)
+		}
+		return RouteRecord{}, fmt.Errorf("get-route user client get failed: %w", err) // untested
+	}
+
```

Comment:
It looks like between this story and #512 we have ended up implementing #640 too: was this a problem with the order of the stories being wrong or stories not being sufficiently decoupled? If not, we should be careful not to introduce too many changes outside the scope of the story.

Reply:
We think this was a problem of ordering--in order to enforce access to a route before modifying it, it made most sense to add access checks to the get method in the route repository (which is also called from the destinations endpoint.) Avoiding this overlap would have required us to do unnatural things in the implementation.

Reply:
right, makes sense - maybe it makes sense to always prioritise `GET`s on every specific resource, it looks like it often ends up overlapping with the other endpoints